app/static/js/update_demand_graph.js

```diff
@@ -0,0 +1,141 @@
+                // calculate the demand by multiplying ts with number of consumers and add it to the total demand
+                var newDemand = data.timeseries_values.map(x => x*nrConsumers);
+                console.log('updating demand from ' + timeseriesVarName + 'with ' + nrConsumers + 'consumers')
+                updateGraph(newDemand, consumerGroupDemandName);
+                },
+            error: function() {
+                    console.error(data);
+                },
+            });
+});
+
+$(document).on('click keyup', '#id_number_consumers', function(){
+// if only number of consumers is changed, retrieve timeseries values from variable and multiply them by new consumers
+        var parentTr = $(this).closest('tr');
+        var parentTrId = parentTr.attr('id');
+        var nrConsumers = parentTr.find('#id_number_consumers').val();
+        var uniqueId = parentTrId.replace('consumer-group', '');
+        var timeseriesVarName = 'timeseriesData_' + uniqueId;
+        var consumerGroupDemandName = 'consumerGroupDemand' + uniqueId;
+        console.log('nr consumers changed :' + nrConsumers + ' consumers')
+            // if timeseries data for this form is saved in the timeseriesdata variable, calculate the new demand for this group
+            if (timeseriesData[timeseriesVarName] !== undefined) {
+                var newDemand = timeseriesData[timeseriesVarName].map(x => x*nrConsumers);
+                console.log('updating demand from ' + timeseriesVarName + 'with ' + nrConsumers + 'consumers')
+                updateGraph(newDemand, consumerGroupDemandName)
+                }
+            });
+
+
+// generate x values for the plot
+// TODO these should be replaced with timestamps
+function generateXValues(start, end) {
+    var xValues = [];
+    for (var i = start; i <= end; i++) {
+        xValues.push(i);
+    }
+    return xValues;
+}
+var startX = 0;
+var endX = 8759;
+var xValues = generateXValues(startX, endX);
+
+
+// create initial graph
+function initialPlot () {
+        console.log('creating initial plot')
+        var plot_div = document.getElementById('demand-aggregate');
+        var plot_data = [{
+            x: xValues,
+            y: [],
+            stackgroup: 'one'
+                }];
+
+        var layout = {
+            xaxis: {range: [0, 8760], title: "Time"},
+            yaxis: {autorange: true, title: "Wh"},
+            title: "Total demand"
+            };
+
+        Plotly.newPlot(plot_div, plot_data, layout)
+        }
+
+
+// data variable to check which traces are already in the graph
+var data = []
+// update graph
+function updateGraph (newDemand, consumerGroupDemandName){
+        console.log('updating graph for consumergroup: ' + consumerGroupDemandName)
+        var plot_div = document.getElementById('demand-aggregate');
+        var trace = {
+            x: xValues,
+            y: newDemand,
+            stackgroup: 'one',
+            name: consumerGroupDemandName
```

Comment:
we should assign this to the `meta` [attribute](https://plotly.com/python-api-reference/generated/plotly.graph_objects.Scatter.html) and reserve `name` for the name which will be displayed on the graph legends